Clarify role of `--system` flag in README (#4031)
astral-sh/uv: README.md

### Installing into arbitrary Python environments

Since uv has no dependency on Python, it can install into virtual environments other than
its own. For example, setting `VIRTUAL_ENV=/path/to/venv` will cause uv to install into
`/path/to/venv`, regardless of where uv is installed. Note that if `VIRTUAL_ENV` is set to
a directory that is **not** a [PEP 405 compliant](https://peps.python.org/pep-0405/#specification)
virtual environment, it will be ignored.

> uv can also install into arbitrary, even non-virtual environments, with the `--python` argument

Reviewer: This is confusing because `--system` is _required_ for this now, right?

Author: No, if it's explicit (i.e. that path to the executable or its directory) we do not require the `--system` flag.

provided to `uv pip sync` or `uv pip install`. For example, `uv pip install --python=/path/to/python`
will install into the environment linked to the `/path/to/python` interpreter.

For convenience, `uv pip install --system` will install into the system Python environment.
Using `--system` is roughly equivalent to `uv pip install --python=$(which python)`,

> but note it will skip `python` executables on the `PATH` that execute in a virtual environment.

Reviewer: Nit: "but note that"

Reviewer: Could "executables on the `PATH` that execute in a virtual environment." be something like... "executables that are linked to virtual environments"?

Author: ```suggestion
but note that executables that are linked to virtual environments will be skipped.
```

Although we generally recommend using virtual environments for dependency management,
`--system` is appropriate in continuous integration and containerized environments.

> The `--system` flag is also used to opt in to mutating system environments. For example, the
> the `--python` argument can be used to request a Python version e.g. `--python 3.12` and uv will
> search for an interpreter that meets the request. If uv finds a system interpreter, e.g., `/usr/lib/python3.12`,
> then the `--system` flag is required to allow modification of this non-virtual Python environment.
> Without the `--system` flag, uv will ignore any interpreters that are not in virtual environments.
> Conversely, when the `--system` flag is provided, uv will ignore any interpreters that are
> in virtual environments.

Reviewer 2: ```suggestion
The `--system` flag is also used to opt in to mutating system environments. For example, the
the `--python` argument can be used to request a Python version (e.g., `--python 3.12`), and uv will
search for an interpreter that meets the request. If uv finds a system interpreter (e.g., `/usr/lib/python3.12`),
then the `--system` flag is required to allow modification of this non-virtual Python environment.
Without the `--system` flag, uv will ignore any interpreters that are not in virtual environments.
Conversely, when the `--system` flag is provided, uv will ignore any interpreters that *are*
in virtual environments.
```
